src/e84_geoai_common/util.py

```diff
@@ -195,3 +195,39 @@ def unique_by[T, K](
             yield item
         elif duplicate_handler_fn:
             duplicate_handler_fn(item, key)
+
+
+def group_by[T, K](
+    items: Iterable[T], *, key_fn: Callable[[T], K] = lambda x: x
+) -> dict[K, list[T]]:
+    """Group items from an iterable by keys produced by a key function.
+
```

Comment:
There is an [itertools.groupby()](https://docs.python.org/3/library/itertools.html#itertools.groupby).